File: KotlinTest/App/src/main/java/com/example/administrator/kotlintest/picture/GlideImageLoader.java
package com.example.administrator.kotlintest.picture;

import android.app.Activity;
import android.net.Uri;
import android.widget.ImageView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.engine.DiskCacheStrategy;
import com.bumptech.glide.request.RequestOptions;
import com.example.administrator.kotlintest.R;
import com.lzy.imagepicker.loader.ImageLoader;

import java.io.File;

/**
 * Created by yechao on 2017/4/28.
 * Describe :
 */

public class GlideImageLoader implements ImageLoader {

    @Override
    public void displayImage(Activity activity, String path, ImageView imageView, int width, int height) {
        RequestOptions options = new RequestOptions();
        options.error(R.mipmap.default_image);
        options.placeholder(R.mipmap.default_image);
        options.diskCacheStrategy(DiskCacheStrategy.ALL);
//        Glide.with(activity)                             //配置上下文
//                .load(Uri.fromFile(new File(path)))      //设置图片路径(fix #8,文件名包含%符号 无法识别和显示)
//                .error(R.mipmap.default_image)           //设置错误图片
//                .placeholder(R.mipmap.default_image)     //设置占位图片
//                .diskCacheStrategy(DiskCacheStrategy.ALL)//缓存全尺寸
//                .into(imageView);

        Glide.with(activity)
                .load(Uri.fromFile(new File(path)))
                .apply(options)
                .into(imageView);
    }

    @Override
    public void clearMemoryCache() {
        //这里是清除缓存的方法,根据需要自己实现
    }
}

File: KotlinTest/App/src/main/java/com/example/administrator/kotlintest/banner/GlideImageLoader.kt
package com.xfs.fsyuncai.ui.main.home.banner

import android.content.Context
import android.net.Uri
import android.widget.ImageView
import com.bumptech.glide.Glide
import com.xfs.fsyuncai.entity.BannerEntity

import com.youth.banner.loader.ImageLoader

class GlideImageLoader : ImageLoader() {
    override fun displayImage(context: Context, path: Any, imageView: ImageView) {

        path?.apply {
//            val entity = path as BannerEntity.DataBean
            Glide.with(context)
//                    .setDefaultRequestOptions(RequestOptions().skipMemoryCache(true))//设置不缓存
                    .load(path)
                    .into(imageView)
//            val uri: Uri = Uri.parse(path)
//            imageView.setImageURI(uri)
        }
    }
}